tests/python_package_test/test_basic.py

```diff
@@ -579,3 +579,30 @@ def test_param_aliases():
     assert all(len(i) >= 1 for i in aliases.values())
     assert all(k in v for k, v in aliases.items())
     assert lgb.basic._ConfigAliases.get('config', 'task') == {'config', 'config_file', 'task', 'task_type'}
+
+
+def _bad_gradients(labels, preds):
+    return np.random.randn(len(labels) + 1), np.random.rand(len(labels) + 1)
+
+
+def _good_gradients(labels, preds):
+    return np.random.randn(len(labels)), np.random.rand(len(labels))
+
+
+def test_custom_objective_safety():
+    X = np.random.randn(100, 5)
+    y_binary = np.random.choice([0, 1], 100)
+    classes = [0, 1, 2]
+    y_multiclass = np.random.choice(classes, 100)
```

Comment:
```suggestion
    X = np.random.randn(100, 5)
    y_binary = np.array([0] * 50 + [1] * 50)
    classes = [0, 1, 2]
    y_multiclass = np.random.choice([0] * 33 + [1] * 33 + [2] * 34)
```

Sorry, just thought of this...can you please remove the randomness from this data construction? Since you're using completely-random data and not testing the produced models, the values don't really matter.

Choosing randomly and using such a small amount of data makes it possible that these tests could fail randomly due to situations like "`y_binary` is all 0s". It may seem like a small probability, but consider that the Python tests run about 40 times on every commit to every pull request in this project.